packages/dashboards/addon/routes/dashboards/dashboard/view.js

```diff
@@ -92,12 +92,36 @@ export default Route.extend({
      * empty filters are pruned from the request
      */
     if (cachedWidgetData && (wasEmptyFilterAdded || wasEmptyFilterRemoved)) {
-      return { dashboard, dataForWidget: cachedWidgetData };
+      return { dashboard, taskByWidget: cachedWidgetData };
     }
-    const dataForWidget = await this.get('dashboardData').fetchDataForDashboard(dashboard);
-    this.set('_widgetDataCache', dataForWidget);
 
-    return { dashboard, dataForWidget };
+    /**
+     * if we don't await here, when going back and forth between dashboards while data is still loading
+     * `deactivate` gets called after these lines, cancelling new tasks and setting the new cache to null
```

Comment:
Do we have testing around this issue?

Reply:
Umm don't think so, will see if I can write one quickly, would be great to find out why it happens too. To be clear it's not related to this PR

Reply:
ok it's a little tricky to navigate directly between 2 dashboards but I've added a test that does this via the index route and also verifies cancelation. Will look further to find out why that happens as route/model does not await the requests anyway